internal/broker/broker_test.go

```diff
@@ -12,6 +12,7 @@ import (
 	"github.com/stretchr/testify/require"
 	"github.com/ubuntu/authd-oidc-brokers/internal/broker"
 	"github.com/ubuntu/authd-oidc-brokers/internal/broker/authmodes"
+	"github.com/ubuntu/authd-oidc-brokers/internal/golden"
```

Comment:
I can see the appeal here, but can you make it a subpackage of `testutils`? I don't think this belongs solely on `internal/`. This way, we can still relate to it as `golden.CheckOrUpdate`, but at least it's organized better.